sormas-backend/src/main/java/de/symeda/sormas/backend/report/AggregateReportFacadeEjb.java

```diff
@@ -312,6 +353,87 @@ public List<AggregatedCaseCountDto> getIndexList(AggregateReportCriteria criteri
 		return resultList;
 	}
 
+	private List<AggregatedCaseCountDto> summarizeAggregateData(
+		final AggregateReportGroupingLevel finalGroupingLevel,
+		List<AggregatedCaseCountDto> queryResult) {
+		final List<AggregatedCaseCountDto> resultList = new ArrayList<>();
+		final Map<AggregatedCaseCountDto, List<AggregatedCaseCountDto>> reportsToBeSummed = new HashMap<>();
+
+		// extract relevant aggregate data and prepare what needs to be sumed
+		for (AggregatedCaseCountDto dto : queryResult) {
+
+			final Optional<AggregatedCaseCountDto> optionalDto = resultList.stream().filter(a -> dto.similar(a, finalGroupingLevel)).findAny();
```

Comment:
Would be faster to sort the queryresults by jurisdiction and only check the last resultList entry for similarity instead of always going through the whole list.
We aren't working with too many results here, so it's probably not worth to rewrite this, but maybe worth a comment so that if needed we can easily optimize the performance here.